src/storage/src/chunk.rs

```diff
@@ -270,7 +276,12 @@ impl ChunkReaderBuilder {
     /// Build time range predicate from schema and filters.
     pub fn build_time_range_predicate(&self) -> TimestampRange {
         let Some(ts_col) = self.schema.user_schema().timestamp_column() else { return TimestampRange::min_to_max() };
-        TimeRangePredicateBuilder::new(&ts_col.name, &self.filters).build()
+        let unit = ts_col
+            .data_type
+            .as_timestamp()
+            .expect("Timestamp column must have timestamp-compatible type")
```

Comment:
question: What if the data type of the timestamp column is Int64?

Reply:
> question: What if the data type of the timestamp column is Int64?

Int64 is treated as `TimestampMillisecond`.

https://github.com/GreptimeTeam/greptimedb/blob/30472cebaeee399ce163780ddb9714806c012ebf/src/datatypes/src/data_type.rs#L177